.github/workflows/release-process.yml

```diff
@@ -0,0 +1,92 @@
+name: Release process
+
+on:
+  workflow_dispatch:
+    inputs:
+      semverVersion:
+        description: 'The semver version used for this release'
+        required: true
+        default: 'undefined'
+        type: string
+      commitHash:
+        description: 'The SHA1 of the commit from which the release is made'
+        required: true
+        default: 'undefined'
+        type: string
+
+jobs:
+  test-and-release-image:
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.commitHash }}
+
+      - name: Set up JDK 21
+        uses: actions/setup-java@v4
+        with:
+          java-version: 21
+          distribution: 'oracle'
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Store short commit hash
+        run: echo "SHORT_COMMIT_HASH=$(git rev-parse --short ${{ inputs.commitHash }})" >> $GITHUB_ENV
+
+      - name: Build JAR with mvn package
+        run: mvn clean package -Dtest=\!CypressE2ETest,\!DockerCypressE2ETest
+
+      - name: Build the docker image
+        run: docker build . -t harbor.puzzle.ch/pitc-mobility-public/uni-luzern:${{ inputs.semverVersion }}.${{ env.SHORT_COMMIT_HASH }}
+
+      - name: Run docker image
+        run: docker run --rm -v ./static:/resources -p 8080:8080 harbor.puzzle.ch/pitc-mobility-public/uni-luzern:${{ inputs.semverVersion }}.${{ env.SHORT_COMMIT_HASH }} &
```

Comment:
It may be possible to use `Testcontainers` + `Spring` to run the application. You can use [this example (Selenium)](https://github.com/testcontainers/testcontainers-java/blob/main/examples/selenium-container/src/test/java/SeleniumContainerTest.java) for inspiration. No need to start a container manually in this workflow!